prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -1221,6 +1252,16 @@ class Hl7Serializer(
             hl7SegmentDelimiter
     }
 
+    fun addLeadingZero(value: String): String {
+        var withLeadingZero: String
+        when (value.length) {
+            3 -> withLeadingZero = "00" + value
+            4 -> withLeadingZero = "0" + value
+            else -> withLeadingZero = value
+        }
+        return withLeadingZero
+    }
+
```

Comment:
@jorg3lopez you should change this to use the built-in `padStart` method that Kotlin provides: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/pad-start.html

I would make it look like this:
```kotlin
fun addLeadingZeros(value: String, padValue: String = "0", padLength: Int = 5): String =>
    value.padStart(padLength, padValue)
```

Reply:
Thanks, Mo. That's is a really nice kotlin feature, I'll make use of it.